### PR TITLE
Slides + lesson R script for Thursday

### PR DESCRIPTION
**Pull request by diazrenata**
@Aariq This is my plan for Thursday!

### CODE REVIEW

File: lessons/8-intermediate-r-2/lesson_script.R [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    
+    
+    island_dat_means <- island_dat |>
+      group_by(species, sex) |>
+      summarize(across(ends_with("mm") | ends_with("g"), mean))
+    
+    island_dat_means_imperial <- island_dat_means |>
+      mutate(
+        across(ends_with("mm"), ~ . * 0.03937008, .names = "{.col}_in"),
+        across(ends_with("g"), ~ . * 0.002204623, .names = "{.col}_lb")
```
Comment (Aariq) [inline lessons/8-intermediate-r-2/lesson_script.R:21]: I can't remember, did you cover using `.names` in `across()`?  If not, might be better to simplify this example and not introduce too many new concepts at once.  If this isn't new, then it's a good opportunity to pause and review.

File: lessons/8-intermediate-r-2/lesson_script.R [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      )
+    
+    island_dat_means_imperial <- island_dat_means_imperial |>
+      rename_with( ~ stringr::str_replace(., "mm_in", "in"), .cols = ends_with("mm_in")) |>
```
Comment (Aariq) [inline lessons/8-intermediate-r-2/lesson_script.R:25]: Does this work?
```suggestion
      rename_with(\(x) stringr::str_replace(x, "mm_in", "in"), .cols = ends_with("mm_in")) |>
```
I think it does, and if it does it might be better to stay consistent with how we show anonymous functions since this is the way you use them in `map()` later on.

File: lessons/8-intermediate-r-2/lesson_script.R [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  island_dat <- island_dat |>
+    na.omit()
+  
+  island_dat_means <- island_dat |>
+    group_by(species, sex) |>
+    summarize(across(ends_with("mm") | ends_with("g"), mean))
+  
+  island_dat_means_imperial <- island_dat_means |>
+    mutate(
+      across(ends_with("mm"), ~ . * 0.03937008, .names = "{.col}_in"),
```
Comment (Aariq) [inline lessons/8-intermediate-r-2/lesson_script.R:79]:
```suggestion
      across(ends_with("mm"), \(x) x * 0.03937008, .names = "{.col}_in"),
```

File: lessons/8-intermediate-r-2/lesson_script.R [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      across(ends_with("g"), ~ . * 0.002204623, .names = "{.col}_lb")
+    )
+  
+  island_dat_means_imperial <- island_dat_means_imperial |>
+    rename_with( ~ stringr::str_replace(., "mm_in", "in"), .cols = ends_with("mm_in")) |>
```
Comment (Aariq) [inline lessons/8-intermediate-r-2/lesson_script.R:84]:
```suggestion
    rename_with(\(x) stringr::str_replace(x, "mm_in", "in"), .cols = ends_with("mm_in")) |>
```

**Comment by Aariq** [comment]
Looks like a good plan!  If you haven't done so yet, might be good to be fairly explicit when introducing anonymous functions.  And it might be easier for students to understand if you use `function(x) x * 10` or `\(x) x* 10` instead of the tidyverse-specific `~ . * 10` notation.

File: lessons/8-intermediate-r-2/lesson_script.R [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    island_dat_means_imperial <- island_dat_means_imperial |>
+      rename_with( ~ stringr::str_replace(., "mm_in", "in"), .cols = ends_with("mm_in")) |>
+      rename_with( ~ stringr::str_replace(., "g_lb", "lb"), .cols = ends_with("g_lb"))
+    
+    island_dat_means_imperial <- island_dat_means_imperial |>
+      select(c(where(is.character) |
+                 ends_with("in")))
+    
+    
+    return(island_dat_means_imperial)
```
Comment (Aariq) [inline lessons/8-intermediate-r-2/lesson_script.R:33]: Technically, `return()` isn't necessary here.  Sometimes I use it just to make it obvious what is being returned, but this works the same
```suggestion
    island_dat_means_imperial
```

File: lessons/8-intermediate-r-2/lesson_script.R (hunk 1 above)
Comment (diazrenata) [reply lessons/8-intermediate-r-2/lesson_script.R:21]: Ah, thanks for reviewing the function! We worked through the process of writing this function on Tuesday, so this whole chunk is actually copy-pasting from the script we wrote on Tuesday. It's a good note about how things might be simplified next time, though!

File: lessons/8-intermediate-r-2/lesson_script.R (hunk 5 above)
Comment (diazrenata) [reply lessons/8-intermediate-r-2/lesson_script.R:33]: Absolutely. I kept `return` in there as I was teaching so that it's explicitly spelled out. We could certainly mention that it's not necessary when we pick it back up.

File: lessons/8-intermediate-r-2/lesson_script.R (hunk 2 above)
Comment (diazrenata) [reply lessons/8-intermediate-r-2/lesson_script.R:25]: This works!

**Comment by diazrenata** [comment]
Thanks for going over the function! I've marked things as resolved because I'm planning to copy-paste the function over again and not spend a lot of time re-writing it. 

I think next time this gets taught it'll be important to think about how we approach passing functions to tidyverse functions (across, map). In this iteration, I demonstrated these functions but didn't go into the concept of writing an anonymous function, because I didn't think the room had the context on writing their own functions yet. In future it may be more tractable for students to learn about _functions_ first, and then learn about functions-that-take-functions in a later session. 🤔 

So, tl;dr I suggest holding off on making changes to the function script for this lesson, but want to think more about how this section of the curriculum fits together for next time!

**Comment by Aariq** [comment]
> Thanks for going over the function! I've marked things as resolved because I'm planning to copy-paste the function over again and not spend a lot of time re-writing it.
> 
Ah, makes sense. I didn't quite understand. I'm impressed at what you guys did last time then! I agree that you don't need to edit it then.  The plan going forward looks good

**Review by Aariq — APPROVED** [review]
